modules/build/src/main/scala/scala/build/preprocessing/ScalaPreprocessor.scala

```diff
@@ -247,8 +247,8 @@ case object ScalaPreprocessor extends Preprocessor {
       val toFilePos = Position.Raw.filePos(path, content)
       val deps = value {
         dependencyTrees
-          .map { t =>
-            val pos      = toFilePos(Position.Raw(t.start, t.end))
+          .map { t => // add 6 to start position to skip ivy syntax ($ivy.`) and dep ($dep.`)
+            val pos      = toFilePos(Position.Raw(t.start + 6, t.end))
```

Comment:
NIT This can be made explicit:
```suggestion
          .map { t => // skip ivy ($ivy.`) or dep syntax ($dep.`)
            val pos      = toFilePos(Position.Raw(t.start + "$ivy.`".length, t.end))
```